c/src/atom.rs

```diff
@@ -303,3 +349,36 @@ impl Drop for CGrounded {
     }
 }
 
+#[cfg(test)]
+mod tests {
+use super::*;
+    #[test]
+    pub fn test_match_callback(){
```

Comment:
Test can be made more compact:
```rust
    #[test]
    pub fn test_match_callback(){
        let var = str_as_cstr("var#1");
        let atom = atom_to_ptr(Atom::sym("atom_test"));
        let vec = vec![ binding_t{ var: var.as_ptr(), atom } ];
        let bindings = (&vec).into();
        
        let mut results: Vec<Bindings> = Vec::new();
        let context = addr_of_mut!(results).cast::<c_void>();

        CGrounded::match_callback(bindings, context);

        assert_eq!(results, vec![Bindings::from(vec![
                (VariableAtom::new_id("var", 1), Atom::sym("atom_test"))])]);
        unsafe{ atom_free(atom) };
    }
```
Also I would add C API test for the custom `match_` implementation instead of Rust one. The C API test will check all the flow front-to-back.

Reply:
Done for Rust part.